### PR TITLE
Replacing gRPC with REST API for rfc2544

### PR DESCRIPTION
**Pull request by dkosteck**
Removal of gRPC code and protobufs, replaced with REST implementation and schema. Using Flask for REST, Marshmallow for serialization, and waitress for production WSGI server. 

Documentation has been updated with examples of the REST API, client removed as curl examples should suffice (although a client can be added if needed).

### CODE REVIEW

File: rfc2544/README.md [hunk 1]
```diff
@@ -49,27 +49,53 @@ podman pod create -p 50051:50051 -n trafficgen
 podman run -d --rm --privileged -v /dev:/dev -v /sys:/sys -v /lib/modules:/lib/modules --cpuset-cpus 4-11 --pod trafficgen -e pci_list=0000:03:00.0,0000:03:00.1  docker.io/cscojianzhan/trafficgen /root/trafficgen_entry.sh server
```
Comment (jianzzha) [inline rfc2544/README.md]: Now that  CMD ["/root/trafficgen_entry.sh", "server"] is used, the README should reflect that, 1) how to kick off a manual run with "/root/trafficgen_entry.sh start" 2) how to kick off the rest api server without the extra argument

**Comment by jianzzha** [comment]
Function wise it looks good.

Let's see if we can tune the test performance up.

1) For reference, here is a set of good manual test options that work well,

```
./binary-search.py \
            --traffic-generator=trex-txrx \
            --device-pairs=${device_pairs} \
            --active-device-pairs=${device_pairs} \
            --sniff-runtime=${sniff_seconds} \
            --search-runtime=${search_seconds} --validation-runtime=${validation_seconds} --max-loss-pct=${loss_ratio} \
            --traffic-direction=bidirectional --frame-size=${size} --num-flows=${flows} \
            --rate-tolerance-failure=fail --duplicate-packet-failure=retry-to-fail --negative-packet-loss=retry-to-fail \
            --send-teaching-warmup \
            --teaching-warmup-packet-type=generic \
            --teaching-warmup-packet-rate=10000 \
            --use-src-ip-flows=1 --use-dst-ip-flows=1 \
            --use-src-mac-flows=0 --use-dst-mac-flows=0 \
            --rate-unit=% --rate=${rate} \
            --one-shot=${steady_rate} \
            --rate-tolerance=10 --runtime-tolerance=10
```

Could you make the missing options available via the POST? for example, the --send-teaching-warmup, --teaching-warmup-packet-type etc?

2) when kicking off the binary search from the python with the subprocess, add "tastset" in front of the binary search commands, - I suspect doing this alone on the entry point script is not enough.

**Comment by jianzzha** [comment]
@dkosteck actually it appears that waitress-serve abd binary search are both running on the 3rd core, as expected. So please ignore the above 2) point

**Comment by akiselev1** [comment]
I assume Github CI workflow should target functionality brought by this PR. Can we keep py client for CI test purposes (maybe in a separate dir)?

**Comment by jianzzha** [comment]
@dkosteck To answer AK's CI question, can you also add a sample py client run the test and print out the result

File: rfc2544/README.md (hunk 1 above)
Comment (dkosteck) [reply rfc2544/README.md]: Updated in latest commit

**Comment by jianzzha** [comment]
@dkosteck Please test out this input (the last commit failed on it):
```
curl -X POST http://localhost:8080/trafficgen/start -H 'Content-Type: application/json' -d '{
    "l3":false,
    "device_pairs":"0:1",
    "search_runtime": 10,
    "validation_runtime":30,
    "num_flows":1000,
    "frame_size":64,
    "max_loss_pct":0.002,
    "sniff_runtime":10,
    "search_granularity":5.0, 
    "teaching_warmup_packet_type":"generic", 
    "teaching_warmup_packet_rate":10000, 
    "use_src_ip_flows":1, 
    "use_dst_ip_flows":1, 
    "use_src_mac_flows":0, 
    "use_dst_mac_flows":0,
    "send_teaching_warmup": true,
    "binary_search_extra_args":["no-promisc"]
}'
```

**Comment by dkosteck** [comment]
@jianzzha Please try again with the latest commit, it should fix the server error. Note that I need to run it with some extra fields to get full line rate (specifically "rate_tolerance":50, "runtime_tolerance":50, "negative_packet_loss":"fail", "rate_tolerance_failure":"fail")

**Review by jianzzha — APPROVED** [review]

**Pull request merged**